libminifi/src/properties/Properties.cpp

```diff
@@ -87,13 +128,17 @@ void Properties::loadConfigureFile(const std::filesystem::path& configuration_fi
     return;
   }
   properties_.clear();
+  dirty_ = false;
   for (const auto& line : PropertiesFile{file}) {
+    auto key = line.getKey();
     auto persisted_value = line.getValue();
     auto value = utils::StringUtils::replaceEnvironmentVariables(persisted_value);
-    properties_[line.getKey()] = {persisted_value, value, false};
+    bool need_to_persist_new_value = false;
+    formatConfigurationProperty(key, persisted_value, value, need_to_persist_new_value);
+    dirty_ |= need_to_persist_new_value;
```

Comment:
This operator does bitwise OR, not logical OR. It happens to work out fine, due to the memory representation of booleans, but I'd prefer using logical operators with booleans.

Reply:
sure, it will be more readable changed it in https://github.com/apache/nifi-minifi-cpp/pull/1543/commits/ee61abd3c1c0efc9f2d460bd691f854f26e9a5aa#diff-f841426e292e70012a81b0fae92f08df7b13120e17b17a92fc360f2f50ec072fR190